src/map/status.c

```diff
@@ -10507,273 +10507,273 @@ static void status_change_start_stop_action(struct block_list *bl, enum sc_type
+		break;
+	case SC_FOOD_DEX_CASH:
+		status_change_end(bl, SC_FOOD_DEX, INVALID_TIMER);
+		status_change_end(bl, SC_FOOD_DEX_CASH, INVALID_TIMER);
+		break;
+	case SC_FOOD_LUK_CASH:
+		status_change_end(bl, SC_FOOD_LUK, INVALID_TIMER);
+		status_change_end(bl, SC_FOOD_LUK_CASH, INVALID_TIMER);
+		break;
+	case SC_GM_BATTLE:
+		status_change_end(bl, SC_GM_BATTLE2, INVALID_TIMER);
+		break;
+	case SC_GM_BATTLE2:
+		status_change_end(bl, SC_GM_BATTLE, INVALID_TIMER);
+		break;
+	case SC_ENDURE:
+		if (val4 == 1)
+			status_change_end(bl, SC_LKCONCENTRATION, INVALID_TIMER);
+		break;
+	case SC_FIGHTINGSPIRIT:
+	case SC_OVERED_BOOST:
+		status_change_end(bl, type, INVALID_TIMER); // Remove previous one.
+		break;
+	case SC_MARSHOFABYSS:
+		status_change_end(bl, SC_INCAGI, INVALID_TIMER);
+		status_change_end(bl, SC_WINDWALK, INVALID_TIMER);
+		status_change_end(bl, SC_ATTHASTE_POTION1, INVALID_TIMER);
+		status_change_end(bl, SC_ATTHASTE_POTION2, INVALID_TIMER);
+		status_change_end(bl, SC_ATTHASTE_POTION3, INVALID_TIMER);
+		status_change_end(bl, SC_ATTHASTE_INFINITY, INVALID_TIMER);
+		break;
+	// Group A Status (doesn't overlap)
+	case SC_SWING:
+	case SC_SYMPHONY_LOVE:
+	case SC_MOONLIT_SERENADE:
+	case SC_RUSH_WINDMILL:
+	case SC_ECHOSONG:
+	case SC_HARMONIZE:
+		if (type != SC_SWING) status_change_end(bl, SC_SWING, INVALID_TIMER);
+		if (type != SC_SYMPHONY_LOVE) status_change_end(bl, SC_SYMPHONY_LOVE, INVALID_TIMER);
+		if (type != SC_MOONLIT_SERENADE) status_change_end(bl, SC_MOONLIT_SERENADE, INVALID_TIMER);
+		if (type != SC_RUSH_WINDMILL) status_change_end(bl, SC_RUSH_WINDMILL, INVALID_TIMER);
+		if (type != SC_ECHOSONG) status_change_end(bl, SC_ECHOSONG, INVALID_TIMER);
+		if (type != SC_HARMONIZE) status_change_end(bl, SC_HARMONIZE, INVALID_TIMER);
+		break;
+	// Group B Status
+	case SC_SIREN:
+	case SC_DEEP_SLEEP:
+	case SC_SIRCLEOFNATURE:
+	case SC_LERADS_DEW:
+	case SC_MELODYOFSINK:
+	case SC_BEYOND_OF_WARCRY:
+	case SC_UNLIMITED_HUMMING_VOICE:
+	case SC_GLOOMYDAY:
+	case SC_SONG_OF_MANA:
+	case SC_DANCE_WITH_WUG:
+		if (type != SC_SIREN) status_change_end(bl, SC_SIREN, INVALID_TIMER);
+		if (type != SC_DEEP_SLEEP) status_change_end(bl, SC_DEEP_SLEEP, INVALID_TIMER);
+		if (type != SC_SIRCLEOFNATURE) status_change_end(bl, SC_SIRCLEOFNATURE, INVALID_TIMER);
+		if (type != SC_LERADS_DEW) status_change_end(bl, SC_LERADS_DEW, INVALID_TIMER);
+		if (type != SC_MELODYOFSINK) status_change_end(bl, SC_MELODYOFSINK, INVALID_TIMER);
+		if (type != SC_BEYOND_OF_WARCRY) status_change_end(bl, SC_BEYOND_OF_WARCRY, INVALID_TIMER);
+		if (type != SC_UNLIMITED_HUMMING_VOICE) status_change_end(bl, SC_UNLIMITED_HUMMING_VOICE, INVALID_TIMER);
+		if (type != SC_GLOOMYDAY) status_change_end(bl, SC_GLOOMYDAY, INVALID_TIMER);
+		if (type != SC_SONG_OF_MANA) status_change_end(bl, SC_SONG_OF_MANA, INVALID_TIMER);
+		if (type != SC_DANCE_WITH_WUG) status_change_end(bl, SC_DANCE_WITH_WUG, INVALID_TIMER);
```

Comment:
if really apply hercules coding style isn't these lines should be
```c
		if (type != SC_SIREN)
			status_change_end(bl, SC_SIREN, INVALID_TIMER);
		if (type != SC_DEEP_SLEEP)
			status_change_end(bl, SC_DEEP_SLEEP, INVALID_TIMER);
```

see below, you done this correctly
```c
	case SC_GENTLETOUCH_ENERGYGAIN:
	case SC_GENTLETOUCH_CHANGE:
	case SC_GENTLETOUCH_REVITALIZE:
		if (type != SC_GENTLETOUCH_REVITALIZE)
			status_change_end(bl, SC_GENTLETOUCH_REVITALIZE, INVALID_TIMER);
		if (type != SC_GENTLETOUCH_ENERGYGAIN)
			status_change_end(bl, SC_GENTLETOUCH_ENERGYGAIN, INVALID_TIMER);
		if (type != SC_GENTLETOUCH_CHANGE)
			status_change_end(bl, SC_GENTLETOUCH_CHANGE, INVALID_TIMER);
		break;
```

Reply:
Yay... I overlooked those. Thanks. 👍

Reply:
Fixed.